### PR TITLE
feat: add C implementation for `stats/base/dists/betaprime/stdev`

### LINKED ISSUE #3446: [RFC]: Add C implementation for `@stdlib/stats/base/dists/betaprime/stdev`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/betaprime/stdev`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/betaprime/stdev" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/betaprime/stdev/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by aayush0325**
Resolves #3446

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for `@stdlib/stats/base/dists/betaprime/stdev` along with relevant tests, docs, examples and benchmarks.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3446

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/README.md [hunk 1]
```diff
@@ -147,6 +147,96 @@ for ( i = 0; i < 10; i++ ) {
...
+Returns the standard deviation of a betaprime distribution.
+
+```c
+double out = stdlib_base_dists_betaprime_stdev( 1.0, 3.0 );
+// returns ~0.866
+```
+
+The function accepts the following arguments:
+
+-   **alpha**: `[in] double` first shape parameter
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/README.md]: @aayush0325 Please see https://github.com/stdlib-js/stdlib/pull/3354. Descriptions should end with periods.

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/README.md [hunk 2]
```diff
@@ -147,6 +147,96 @@ for ( i = 0; i < 10; i++ ) {
...
+#include <stdio.h>
+
+int main( void ) {
+    double alpha;
+    double beta;
+    double y;
+    int i;
+
+    for ( i = 0; i < 25; i++ ) {
+        alpha = ( (double)rand() / (double)RAND_MAX ) * 20.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/README.md]: See https://github.com/stdlib-js/stdlib/pull/3354 for random value generation.

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/benchmark/benchmark.native.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+bench( pkg, opts, function benchmark( b ) {
+	var alpha;
+	var beta;
+	var y;
+	var i;
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		alpha = ( randu() * 10.0 ) + EPS;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/benchmark/benchmark.native.js]: See https://github.com/stdlib-js/stdlib/pull/3354 for value generation.

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/benchmark/c/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	double elapsed;
+	double alpha;
+	double beta;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		alpha = 20.0 * rand_double();
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/benchmark/c/benchmark.c]: See https://github.com/stdlib-js/stdlib/pull/3354 for random value generation.

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/examples/c/example.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+#include <stdio.h>
+
+int main( void ) {
+	double alpha;
+	double beta;
+	double y;
+	int i;
+
+	for ( i = 0; i < 25; i++ ) {
+		alpha = ( (double)rand() / (double)RAND_MAX ) * 20.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/examples/c/example.c]: See https://github.com/stdlib-js/stdlib/pull/3354 for an example of random value generation.

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/lib/native.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Returns the standard deviation of a beta prime distribution.
+*
+* @param {PositiveNumber} alpha - first shape parameter
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/lib/native.js:32]: Missing `@private`

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/src/main.c [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/stats/base/dists/betaprime/stdev.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/sqrt.h"
+
+/**
+* Returns the standard deviation of an beta prime distribution.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/src/main.c]:
```suggestion
* Returns the standard deviation of a beta prime distribution.
```

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/src/main.c [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+*
+* @param alpha    first shape parameter
+* @param beta    second shape paramter
+* @return     stdandard deviation
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/src/main.c]: Description is not aligned.

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/benchmark/benchmark.native.js (hunk 3 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/benchmark/benchmark.native.js]: You should make similar changes to the other JS benchmark file.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this, @aayush0325. Spot checking your other PRs, it looks like you need to make similar changes across all of them. Please see https://github.com/stdlib-js/stdlib/pull/3354 for an idea of what we are looking for.

**Comment by aayush0325** [comment]
good evening @kgryte, ive pushed the requested changes in this and all other PRs. thanks for your time!

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/README.md [hunk 9]
```diff
@@ -147,6 +147,101 @@ for ( i = 0; i < 10; i++ ) {
...
+        return min + ( v * ( max - min ) );
+    }
+
+    for ( i = 0; i < 25; i++ ) {
+        alpha = random_uniform( 0, 20 );
+        beta = random_uniform( 0, 20 ) + 2.0;
+        y = stdlib_base_dists_betaprime_stdev( alpha, beta );
+        printf( "α: %lf, β: %lf, SD(X;α,β): %lf\n", alpha, beta, y );
+    }
+}
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/README.md]:
```suggestion

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v * ( max - min ) );
}

int main( void ) {
    double alpha;
    double beta;
    double y;
    int i;

    for ( i = 0; i < 25; i++ ) {
        alpha = random_uniform( 0, 20 );
        beta = random_uniform( 0, 20 ) + 2.0;
        y = stdlib_base_dists_betaprime_stdev( alpha, beta );
        printf( "α: %lf, β: %lf, SD(X;α,β): %lf\n", alpha, beta, y );
    }
}
```

File: lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/README.md [hunk 10]
```diff
@@ -147,6 +147,101 @@ for ( i = 0; i < 10; i++ ) {
...
+
+### Usage
+
+```c
+#include "stdlib/stats/base/dists/betaprime/stdev.h"
+```
+
+#### stdlib_base_dists_betaprime_stdev( alpha, beta )
+
+Returns the standard deviation of a betaprime distribution.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/betaprime/stdev/README.md]:
```suggestion
Returns the standard deviation of a beta prime distribution.
```

**Review by Planeshifter — APPROVED** [review]
Thank you, @aayush0325! Let's get this one in.

**Pull request merged**